imap_processing/glows/l0/glows_l0_data.py

```diff
@@ -152,50 +132,61 @@ class DirectEventL0(GlowsL0):
     MET : int
         CCSDS Packet Time Stamp (coarse time)
     SEC : int
-        Data timestamp, seconds counter.
+        Data IMAP timestamp, seconds counter.
     LEN : int
         Number of packets in data set.
     SEQ : int
         Packet sequence in data set.
+    DE_DATA : bytearray
+        Raw direct event data (compressed)
     ground_sw_version : str
         Ground software version
     packet_file_name : str
         File name of the source packet
     ccsds_header : CcsdsData
         CCSDS header data
+
+    Methods
+    -------
+    sequence_match_check
+        Compare two DirectEventL0 objects and see if MET and LEN are equal
+
     """
 
     MET: int
     SEC: int
     LEN: int
     SEQ: int
+    DE_DATA: bytearray
+
+    def __post_init__(self):
+        """Convert from string to bytearray if DE_DATA is a string of ones and zeros."""
+        if isinstance(self.DE_DATA, str):
+            # Convert string output from space_packet_parser to bytearray
+            self.DE_DATA = bytearray(
+                int(self.DE_DATA, 2).to_bytes(len(self.DE_DATA) // 8, "big")
+            )
+
+    def sequence_match_check(self, de_l0):
+        """
+        Compare fields for L0 which should be the same for packets within one sequence.
 
-    def __init__(self, packet, software_version: str, packet_file_name: str):
-        """Initialize data class with a packet of direct event data.
+        This method compares the IMAP time (MET) and packet length (LEN) fields.
```

Comment:
What is the reason for comparing MET and LEN? Does this mean that there are other packets with same MET and LEN. If they match, then those data belong together to create one full data?

Reply:
Yes, exactly. The direct events data can be split across multiple packets, so this method checks that the packets actually match. I will rename it (like Greg suggested) and update the comment to be a bit more clear.